doc/variableexplorer.rst

```diff
@@ -2,95 +2,235 @@
-Types with specialized editing support include:
+The Variable Explorer's toolbar includes several useful features that affect the entire namespace. For example, you can save the current session's data as a `.spydata` file, which can be loaded later to recover all the variables stored.
+
+.. image:: images/variable_explorer/variable-explorer-import-data.gif
+   :align: center
+   :alt: Variable Explorer showing how to save and import data
+
+|
+
+There is also a button to remove all displayed variables, and a search box to find objects by  name or type.
+
+.. image:: images/variable_explorer/variable-explorer-search.gif
+   :align: center
+   :alt: Variable Explorer showing how to search variables
+
+|
+
+Finally, there is a button to refresh the Variable Explorer's contents, which will update it to show the current state of the code running in the IPython console.
+
+
+======================
+Advanced Functionality
+======================
+
+The context menu, available by right-clicking any variable, provides numerous additional options to interact with objects of various types. These include renaming, removing or editing existing variables, as well as the :guilabel:`duplicate` option to create a new copy of one of them under a new name you enter in the resulting dialog box.
+
+.. image:: images/variable_explorer/variable-explorer-duplicate.gif
+   :align: center
+   :alt: Variable Explorer showing duplicating a variable
+
+| 
+
+Furthermore, you can copy and paste the value of a variable, saving it in the Variable Explorer with any name that you choose. This allows you to change the type of the variable that you are pasting which can be very useful, allowing to, for example, easily copy the elements of a list into an array.
+
+.. image:: images/variable_explorer/variable-explorer-copy-paste.gif
+   :align: center
+   :alt: Variable Explorer showing copying list into array
+
+| 
 
-* Integers
-* Floats
-* Complex numbers
-* Strings
-* ``datetime`` dates and ``Timedelta`` s
-* Lists
-* Tuples
-* Sets
-* Dictionaries
-* NumPy arrays and matrices
-* Pandas ``DataFrame``, ``TimeSeries`` and ``Index`` (any type) objects
-* ``PIL``/``Pillow`` images
-* Namespaces
+Additionally, you can create a variable from scratch directly in the Variable Explorer with the :guilabel:`Insert` option, which allows you to type the key (which should be in quotation marks) and the value for the item that you want to insert. In addition to adding a new top-level variable, this feature also allows you to create a new key in a dictionary, a new element in a list, and much more.
 
+.. image:: images/variable_explorer/variable-explorer-insert.gif
+   :align: center
+   :alt: Variable Explorer showing insertion of a new variable
+
+| 
+
+For lists and Numpy arrays, more advanced options are available, including generating plots and histograms of their values appropriate to their type and dimensions.
+
+.. image:: images/variable_explorer/variable-explorer-histogram-plot.gif
```

Comment:
This is a bit confusing because you have to move from the Variable Explorer to Plots to show the generated images, Instead, I suggest to hide the Editor, show the Variable Explorer on the left half of the entire window and Plots on the right half. That wayt it'd be easier for users to understand what's going on.

Reply:
Done

Reply:
LGTM